#! /usr/bin/env python

PACKAGE='olympe_bridge'

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       	Name					Type      	Level	Description						Default 	Min	Max 	Values
gen.add(	"max_tilt",    			int_t,    	1, 	"Max pitch/roll (in deg)", 				10,  		1, 	40) 	# default = 10
gen.add(	"max_vertical_speed",  		double_t,    	1, 	"Max vertical speed (in m/s)", 			1,  		0.1, 	4) 	# default = 3
gen.add(	"max_horizontal_speed",  		double_t,    	1, 	"Max horizontal speed (in m/s)", 			1,  		0.1, 	15) 	# 
gen.add(	"max_yaw_rotation_speed",  		int_t,    	1, 	"Max yaw rotation speed (in deg/s)", 			100,  		3, 	200) 	# default = 40
gen.add(	"max_pitch_roll_rotation_speed", 	int_t,    	1, 	"Max pitch/roll rotation speed (in deg/s)", 		200,  		40, 	300) 	# default = 116
gen.add(	"max_distance",    			int_t,    	1, 	"Max distance (in m)", 				10,  		10, 	4000) 	# default = 100
gen.add(	"max_altitude",    			double_t,    	1, 	"Max altitude (in m)", 				2,  		0.5, 	4000) 	# default = 150
gen.add(	"banked_turn",    			bool_t,    	1, 	"Enable/Disable banked turn", 			True)				# default = True
gen.add(	"gimbal_compensation",    		bool_t,    	2, 	"Enable/Disable gimbal roll/pitch compensation", 	True)  			# default = True
gen.add(	"max_gimbal_speed",    		int_t,    	2, 	"Max gimbal speed (in deg/s)", 			180,  		1, 	180)  	# default = 

exit(gen.generate(PACKAGE, "olympe_bridge", "setAnafi"))
